demos/faults.cc

```diff
@@ -0,0 +1,62 @@
+/*
```

Comment:
This file is tricky, is it a reasonable request to ask for unit tests? (Thinking of like three tests, one which sends no signal, one which sends SIGSEGV, and one which sends some other signal.)

Reply:
Definitely reasonable. Tests added in next revision.